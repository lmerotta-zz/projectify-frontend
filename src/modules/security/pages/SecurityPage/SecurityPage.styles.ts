import { Box, Grid, styled, Typography } from "@mui/material";
import background from "../../images/background.svg";

export const Wrapper = styled(Grid)({
  minHeight: "100vh",
});

export const Hero = styled(Grid)(({ theme }) => ({
  paddingLeft: theme.spacing(4),
  paddingRight: theme.spacing(4),
  paddingTop: theme.spacing(10),
  paddingBottom: theme.spacing(10),
  backgroundImage: `url(${background})`,
  backgroundSize: "cover",

  [theme.breakpoints.up("sm")]: {
    paddingTop: theme.spacing(6),
    paddingBottom: theme.spacing(6),
  },

  [theme.breakpoints.up("md")]: {
    paddingLeft: theme.spacing(4),
    paddingRight: theme.spacing(4),
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
}));

export const HeroContent = styled(Box)(({ theme }) => ({
  [theme.breakpoints.up("md")]: {
    width: "66%",
  },
}));

export const Title = styled(Typography)(({ theme }) => ({
  display: "flex",
  justifyContent: "center",
  marginBottom: theme.spacing(6),

  [theme.breakpoints.up("md")]: {
    marginTop: "auto",
    marginBottom: "auto",
  },
})) as typeof Typography;

export const SubTitle = styled(Typography)(({ theme }) => ({
  marginBottom: theme.spacing(1),
  ...(theme.typography.h6 as any),
  fontWeight: "normal",

  [theme.breakpoints.up("md")]: {
    ...(theme.typography.h2 as any),
    fontWeight: "normal",
  },
})) as typeof Typography;

export const Caption = styled(Typography)(({ theme }) => ({
  ...(theme.typography.body2 as any),
  fontWeight: "lighter",

  [theme.breakpoints.up("md")]: {
    ...(theme.typography.subtitle1 as any),
    fontWeight: "lighter",
  },
})) as typeof Typography;

export const RightPane = styled(Grid)(({ theme }) => ({
  padding: theme.spacing(4),
  flexGrow: 1,
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",

  [theme.breakpoints.up("sm")]: {
    paddingLeft: theme.spacing(20),
    paddingRight: theme.spacing(20),
    paddingTop: theme.spacing(5),
    paddingBottom: theme.spacing(5),
  },

  [theme.breakpoints.up("md")]: {
    paddingLeft: theme.spacing(6),
    paddingRight: theme.spacing(6),
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
  },
}));
