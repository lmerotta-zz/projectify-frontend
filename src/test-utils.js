import React from "react";
import { render } from "@testing-library/react";
import { CacheProvider } from "@emotion/react";
import createCache from "@emotion/cache";

const cache = createCache({ key: "iodajdoiwanoawdni" });
cache.compat = true;

const AllTheProviders = ({ children }) => {
  return <CacheProvider value={cache}>{children}</CacheProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
