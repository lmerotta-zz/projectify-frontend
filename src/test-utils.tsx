import { MockedProvider, MockedProviderProps } from "@apollo/client/testing";
import {
  CircularProgress,
  StyledEngineProvider,
  ThemeProvider,
} from "@mui/material";
import { render as realRender, RenderOptions } from "@testing-library/react";
import { ReactElement, ReactNode, Suspense } from "react";
import { MemoryRouter, MemoryRouterProps } from "react-router";
import { Route, Routes } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import theme from "theme";

function renderWithProviders(
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries"> & {
    routerProps?: MemoryRouterProps;
    graphqlProps?: MockedProviderProps;
  }
) {
  const Wrapper = ({ children }: { children?: ReactNode }) => {
    return (
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>
          <ToastContainer
            theme="colored"
            autoClose={3000}
            containerId="react-toastify"
          />
          <MockedProvider {...options?.graphqlProps} addTypename={false}>
            <MemoryRouter {...options?.routerProps}>
              <Suspense fallback={<CircularProgress />}>
                <Routes>
                  <Route
                    path="/*"
                    element={
                      <>
                        <h1>TESTSASATATAST</h1>
                        {children}
                      </>
                    }
                  />
                </Routes>
              </Suspense>
            </MemoryRouter>
          </MockedProvider>
        </ThemeProvider>
      </StyledEngineProvider>
    );
  };

  return realRender(ui, { wrapper: Wrapper, ...options });
}

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { renderWithProviders };
