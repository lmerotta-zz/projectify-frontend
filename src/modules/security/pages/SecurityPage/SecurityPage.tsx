import { useReactiveVar } from "@apollo/client";
import { LinearProgress, Typography } from "@mui/material";
import { isAuthenticated } from "apollo/local-state";
import { AnimatePresence } from "framer-motion";
import { lazy, Suspense } from "react";
import { Trans, useTranslation } from "react-i18next";
import { Navigate, Route, Routes, useLocation } from "react-router-dom";
import { RightPane } from "modules/security/components";
import * as Styles from "./SecurityPage.styles";

const LoginPage = lazy(() => import("modules/security/pages/LoginPage/LoginPage"));
const RegisterPage = lazy(
  () => import("modules/security/pages/RegisterPage/RegisterPage")
);

const SecurityPage = () => {
  const { t } = useTranslation();
  const location = useLocation();
  const authenticated = useReactiveVar(isAuthenticated);

  if (authenticated === null) {
    return null;
  }

  return authenticated ? (
    <Navigate to="/" replace />
  ) : (
    <Styles.Wrapper container spacing={0}>
      <Styles.Hero
        item
        xs={12}
        md={7}
        lg={8}
        container
        direction="column"
        justifyContent="center"
      >
        <Styles.Title variant="h3" component="h2" color="white">
          ProjectifyTmp
        </Styles.Title>
        <Styles.HeroContent>
          <Styles.SubTitle color="white" component="h4">
            <Trans
              i18nKey="security.login_page.hero.subtitle"
              components={{
                highlight: (
                  <Typography
                    color="secondary"
                    component="span"
                    variant="inherit"
                  />
                ),
              }}
            />
          </Styles.SubTitle>
          <Styles.Caption color="white">
            {t("security.login_page.hero.caption")}
          </Styles.Caption>
        </Styles.HeroContent>
      </Styles.Hero>

      <Suspense
        fallback={
          <RightPane>
            <LinearProgress variant="indeterminate" />
          </RightPane>
        }
      >
        <AnimatePresence exitBeforeEnter initial={false}>
          <Routes location={location} key={location.pathname}>
            <Route path="login" element={<LoginPage />} />
            <Route path="register" element={<RegisterPage />} />
            <Route path="*" element={<Navigate to="./login" replace />} />
          </Routes>
        </AnimatePresence>
      </Suspense>
    </Styles.Wrapper>
  );
};

export default SecurityPage;
