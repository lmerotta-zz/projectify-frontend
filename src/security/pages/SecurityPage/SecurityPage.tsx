import { LinearProgress, Typography } from "@mui/material";
import { AnimatePresence } from "framer-motion";
import { lazy, Suspense } from "react";
import { Trans, useTranslation } from "react-i18next";
import { Route, Routes, useLocation } from "react-router-dom";
import { RightPane } from "security/components";
import * as Styles from "./SecurityPage.styles";

const LoginPage = lazy(() => import("security/pages/LoginPage/LoginPage"));
const RegisterPage = lazy(
  () => import("security/pages/RegisterPage/RegisterPage")
);

const SecurityPage = () => {
  const { t } = useTranslation();
  const location = useLocation();

  return (
    <Styles.Wrapper container spacing={0}>
      <Styles.Hero
        item
        xs={12}
        md={7}
        lg={8}
        container
        direction="column"
        justifyContent="center"
      >
        <Styles.Title variant="h3" component="h2" color="white">
          ProjectifyTmp
        </Styles.Title>
        <Styles.HeroContent>
          <Styles.SubTitle color="white" component="h4">
            <Trans
              i18nKey="security.login_page.hero.subtitle"
              components={{
                highlight: (
                  <Typography
                    color="secondary"
                    component="span"
                    variant="inherit"
                  />
                ),
              }}
            />
          </Styles.SubTitle>
          <Styles.Caption color="white">
            {t("security.login_page.hero.caption")}
          </Styles.Caption>
        </Styles.HeroContent>
      </Styles.Hero>

      <Suspense
        fallback={
          <RightPane>
            <LinearProgress variant="indeterminate" />
          </RightPane>
        }
      >
        <AnimatePresence exitBeforeEnter initial={false}>
          <Routes location={location} key={location.pathname}>
            <Route path="login" element={<LoginPage />} />
            <Route path="register" element={<RegisterPage />} />
          </Routes>
        </AnimatePresence>
      </Suspense>
    </Styles.Wrapper>
  );
};

export default SecurityPage;
