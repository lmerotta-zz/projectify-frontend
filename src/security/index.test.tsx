import { render } from "test-utils";
import { MemoryRouter } from "react-router-dom";
import SecurityPage from ".";

describe("Security main page unit tests", () => {
  it("Renders", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SecurityPage />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
