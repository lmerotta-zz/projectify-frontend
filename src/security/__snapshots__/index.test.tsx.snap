// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security main page unit tests Renders 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

@media (min-width: 1024px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-1 {
  width: 100%;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 4rem;
  padding-bottom: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  background-image: url(background.svg);
  -webkit-background-size: cover;
  background-size: cover;
}

.emotion-1>:not([hidden])~:not([hidden]) {
  --tw-space-y-reverse: 0;
  margin-top: calc(3rem * calc(1 - var(--tw-space-y-reverse)));
  margin-bottom: calc(3rem * var(--tw-space-y-reverse));
}

@media (min-width: 1024px) {
  .emotion-1 {
    width: 66.666667%;
  }
}

.emotion-2 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  --tw-text-opacity: 1;
  color: rgba(63, 63, 70, var(--tw-text-opacity));
  font-weight: 700;
  --tw-text-opacity: 1;
  color: rgba(250, 250, 250, var(--tw-text-opacity));
}

@media (min-width: 768px) {
  .emotion-2 {
    font-size: 3rem;
    line-height: 1;
  }
}

@media (min-width: 1024px) {
  .emotion-2 {
    margin-top: auto;
    margin-bottom: auto;
  }
}

@media (min-width: 1024px) {
  .emotion-3 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: flex-start;
    align-self: flex-start;
    width: 66.666667%;
  }
}

.emotion-4 {
  font-size: 1.25rem;
  line-height: 1.75rem;
  --tw-text-opacity: 1;
  color: rgba(161, 161, 170, var(--tw-text-opacity));
  font-weight: 500;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  font-weight: 400;
  margin-bottom: 0.75rem;
}

@media (min-width: 768px) {
  .emotion-4 {
    font-size: 1.5rem;
    line-height: 2rem;
  }
}

@media (min-width: 1024px) {
  .emotion-4 {
    font-size: 3.75rem;
    line-height: 1;
  }
}

.emotion-5 {
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
}

@media (min-width: 1024px) {
  .emotion-5 {
    font-size: 1.25rem;
    line-height: 1.75rem;
  }
}

.emotion-6 {
  width: 100%;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 2.5rem;
  padding-bottom: 2.5rem;
}

@media (min-width: 768px) {
  .emotion-6 {
    padding-left: 12rem;
    padding-right: 12rem;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

@media (min-width: 1024px) {
  .emotion-6 {
    width: 33.333333%;
    padding-left: 4rem;
    padding-right: 4rem;
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <h1
        class="emotion-2"
      >
        ProjectifyTmp
      </h1>
      <div
        class="emotion-3"
      >
        <h2
          class="emotion-4"
        >
          security.login_page.hero.subtitle
        </h2>
        <p
          class="emotion-5"
        >
          security.login_page.hero.caption
        </p>
      </div>
    </div>
    <div
      class="emotion-6"
    />
  </div>
</DocumentFragment>
`;
